Modules/Forum/classes/class.ilObjForumGUI.php

```diff
@@ -2706,26 +2694,28 @@ private function getForumObjects()
 
     public function checkUsersViewMode() : void
     {
-        if (!isset($_SESSION['viewmode'])) {
-            $_SESSION['viewmode'] = $this->objProperties->getDefaultView();
+        if (!isset($this->selectedSorting)) {
+            $this->selectedSorting = $this->objProperties->getDefaultView();
         }
 
-        if (isset($_GET['viewmode']) && (int) $_GET['viewmode'] !== (int) $_SESSION['viewmode']) {
-            $_SESSION['viewmode'] = (int) $_GET['viewmode'];
+        if (isset($this->httpRequest->getQueryParams()['viewmode'])
+            && (int) $this->httpRequest->getQueryParams()['viewmode']  !== (int) $this->selectedSorting) {
+            $this->selectedSorting = (int) $this->httpRequest->getQueryParams()['viewmode'];
         }
 
-        if (isset($_GET['action']) && !in_array($_SESSION['viewmode'], [
+        if (isset($_GET['action']) && !in_array($this->selectedSorting, [
             ilForumProperties::VIEW_TREE,
             ilForumProperties::VIEW_DATE_ASC,
             ilForumProperties::VIEW_DATE_DESC
         ])) {
-            $_SESSION['viewmode'] = $this->objProperties->getDefaultView();
+            $this->selectedSorting = $this->objProperties->getDefaultView();
         }
+        \ilSession::set('viewmode',$this->selectedSorting);
```

Comment:
Where is the `viewmode` fetched by calling `ilSession::set('viewmode')`? I did not find any usage.

* The `viewmode` **MUST** be persisted in session (if changed) and **always** read from session when determining the `effective viewmode` for the presentation of postings.
* If the `viewmode` is changed by an according UI action (by choosing a another **valid** `viewmode`), it **MUST** be modified in the session and used for the presentation of postings.